controllers/numaresourcesoperator_controller_test.go

```diff
@@ -106,36 +106,94 @@ var _ = Describe("Test NUMAResourcesOperator Reconcile", func() {
 
 	Context("with unexpected NRO CR name", func() {
 		It("should updated the CR condition to degraded", func() {
-			nro := testobjs.NewNUMAResourcesOperator("test", nil)
+			nro := testobjs.NewNUMAResourcesOperator("test")
 			verifyDegradedCondition(nro, status.ConditionTypeIncorrectNUMAResourcesOperatorResourceName)
 		})
 	})
 
-	Context("with NRO empty machine config pool selector node group", func() {
-		It("should updated the CR condition to degraded", func() {
-			nro := testobjs.NewNUMAResourcesOperator(objectnames.DefaultNUMAResourcesOperatorCrName, []*metav1.LabelSelector{nil})
+	Context("with NRO empty selectors node group", func() {
+		It("should update the CR condition to degraded", func() {
+			ng := nropv1.NodeGroup{}
+			nro := testobjs.NewNUMAResourcesOperator(objectnames.DefaultNUMAResourcesOperatorCrName, ng)
 			verifyDegradedCondition(nro, validation.NodeGroupsError)
 		})
 	})
 
-	Context("without available machine config pools", func() {
-		It("should updated the CR condition to degraded", func() {
-			nro := testobjs.NewNUMAResourcesOperator(objectnames.DefaultNUMAResourcesOperatorCrName, []*metav1.LabelSelector{
-				{
+	Context("with NRO mutiple pool specifiers set on same node group", func() {
+		It("should update the CR condition to degraded", func() {
+			pn := "pn-1"
+			ng := nropv1.NodeGroup{
+				MachineConfigPoolSelector: &metav1.LabelSelector{
 					MatchLabels: map[string]string{"test": "test"},
 				},
-			})
+				PoolName: &pn,
+			}
+			nro := testobjs.NewNUMAResourcesOperator(objectnames.DefaultNUMAResourcesOperatorCrName, ng)
 			verifyDegradedCondition(nro, validation.NodeGroupsError)
 		})
 	})
 
+	Context("without available machine config pools", func() {
+		It("should update the CR condition to degraded when MachineConfigPoolSelector is set", func() {
+			ng := nropv1.NodeGroup{
+				MachineConfigPoolSelector: &metav1.LabelSelector{MatchLabels: map[string]string{"test": "test"}},
+			}
+			nro := testobjs.NewNUMAResourcesOperator(objectnames.DefaultNUMAResourcesOperatorCrName, ng)
+			verifyDegradedCondition(nro, validation.NodeGroupsError)
+		})
+		//It("should update the CR condition to degraded when PoolName set", func() {
+		//	pn := "pn-1"
+		//	ng := nropv1.NodeGroup{
+		//		PoolName: &pn,
+		//	}
+		//	nro := testobjs.NewNUMAResourcesOperator(objectnames.DefaultNUMAResourcesOperatorCrName, ng)
+		//	verifyDegradedCondition(nro, validation.NodeGroupsError)
+		//})
```

Comment:
likewise, let's fix this commented code

Reply:
sure, will remove it, I added them like this because I wasn't clear what direction we want here: https://github.com/openshift-kni/numaresources-operator/pull/1018#discussion_r1786005282
but now I'll remove them. thanks for the review and comments